src/rgw/rgw_common.h

```diff
@@ -146,6 +146,13 @@ using ceph::crypto::MD5;
 #define ERR_SIGNATURE_NO_MATCH   2027
 #define ERR_INVALID_ACCESS_KEY   2028
 #define ERR_BUCKET_ALREADY_OWNED 2029
+#define ERR_BAD_RENAME_REQ       2030
```

Comment:
User will be given an unlikely error code only when we want him to file a bug. Else he will see "failed in rename" only.